spec/tdd_solver_spec.rb

```diff
@@ -0,0 +1,42 @@
+require_relative '../solver'
+
+describe Solver do
+  before(:each) do
+    @solver = Solver.new
+  end
+
+  context 'testing the factorial method' do
+    it 'returns 1 when given 0' do
+      expect(@solver.factorial(0)).to eql 1
+    end
+
```

Comment:
- [ ]Great job with the tests cases. Your are supposed to have a test case for factorial that raises an exception when a negative number is given as per project requirements.
![image](https://github.com/citec-47/TDD-PROJECT/assets/62849872/f675213f-4d8c-4015-a0a7-82c874cde16c)

Reply:
Thanks! The test case have been added